src/main/scala/org/fortysevendeg/sparksftp/ReadingSFTPHadoopApp.scala

```diff
@@ -26,59 +26,75 @@ object ReadingSFTPHadoopApp extends IOApp {
           "spark.kryo.registrationRequired",
           config.spark.serializer.contains("KryoSerializer").toString
         )
-        .set("spark.hadoop.fs.sftp.impl", "org.apache.hadoop.fs.sftp.SFTPFileSystem")
-        .set("fs.sftp.impl", "org.apache.hadoop.fs.sftp.SFTPFileSystem")
-        .set("fs.sftp.proxy.host", config.sftp.sftpHost)
-        .set("fs.sftp.host.port", "22")
+        .set("fs.sftp.impl", "org.apache.hadoop.fs.sftpwithseek.SFTPFileSystem")
         .set("fs.sftp.impl.disable.cache", "true")
+        .set("spark.sql.legacy.allowCreatingManagedTableUsingNonemptyLocation","true") //https://kb.azuredatabricks.net/jobs/spark-overwrite-cancel.html
         .registerKryoClasses(RegisterInKryo.classes.toArray)
 
-      sftpPath = s"sftp://${config.sftp.sftpUser}:${config.sftp.sftpPass}@${config.sftp.sftpHost}" + s":${config.sftp.sftpPath}"
-
       sparkSession: SparkSession = SparkSession.builder
         .config(defaultSparkConf)
         .enableHiveSupport
         .getOrCreate()
 
-      // Others way of reading, if we wanted to read as inputstreamT
-      // context = sparkSession.sparkContext
-      // _ = context.addFile(sftpPath)
-      // fileName = SparkFiles.get(sftpPath.split("/").last)
-      // fileRDD = context.textFile(fileName)
-      // sourceFS = FileSystem.get(sourceUri, context.hadoopConfiguration)
-      // fsinputStream = sourceFS.open(new Path(sourceUri.getPath))
-      // context.textFile
-      // context.hadoopFile
-      // context.newAPIHadoopFile()
+      sftpUser1 = sparkSession.sparkContext.getConf
+        .getOption("spark.executorEnv.SFTP_USER")
+        .getOrElse(config.sftp.sftpUser)
+      sftpPass1 = sparkSession.sparkContext.getConf
+        .getOption("spark.executorEnv.SFTP_PASS")
+        .getOrElse(config.sftp.sftpPass)
+      sftpHost1 = sparkSession.sparkContext.getConf
+        .getOption("spark.executorEnv.SFTP_HOST")
+        .getOrElse(config.sftp.sftpHost)
+      sftpPath1 = sparkSession.sparkContext.getConf
+        .getOption("spark.executorEnv.SFTP_PATH")
+        .getOrElse(config.sftp.sftpPath)
```

Comment:
Same comments in this file that for the previous one.

Reply:
Changed in: d2734cd84776a08db06bbaa45e87bc268135861c